tools/cli/adminDLQCommands.go

```diff
@@ -107,20 +108,27 @@ func AdminPurgeDLQMessages(c *cli.Context) {
 	defer cancel()
 
 	dlqType := getRequiredOption(c, FlagDLQType)
-	sourceCluster := getRequiredOption(c, FlagTargetCluster)
-	shardID := getRequiredIntOption(c, FlagShardID)
-	lastMessageID := getRequiredInt64Option(c, FlagLastMessageID)
+	sourceCluster := getRequiredOption(c, FlagSourceCluster)
+	lowerShardBound := c.Int(FlagLowerShardBound)
+	upperShardBound := c.Int(FlagUpperShardBound)
+	var lastMessageID *int64
+	if c.IsSet(FlagLastMessageID) {
+		lastMessageID = common.Int64Ptr(c.Int64(FlagLastMessageID))
+	}
 
 	adminClient := cFactory.ServerAdminClient(c)
-	if err := adminClient.PurgeDLQMessages(ctx, &replicator.PurgeDLQMessagesRequest{
-		Type:                  toQueueType(dlqType),
-		SourceCluster:         common.StringPtr(sourceCluster),
-		ShardID:               common.Int32Ptr(int32(shardID)),
-		InclusiveEndMessageID: common.Int64Ptr(lastMessageID),
-	}); err != nil {
-		ErrorAndExit("Failed to purge dlq", err)
+	for shardID := lowerShardBound; shardID <= upperShardBound; shardID++ {
+		if err := adminClient.PurgeDLQMessages(ctx, &replicator.PurgeDLQMessagesRequest{
```

Comment:
If there's a large number of shards, default context timeout may be too small for all the requests? Maybe create a new context for each request? Same question for merge command.

Reply:
Updated